src/libs/hashCode.js

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable no-bitwise */
+/**
+ * Simple string hashing function obtained from: https://stackoverflow.com/a/8831937/16434681
+ * Returns a hash code from a string
+ * @param  {String} str The string to hash.
+ * @return {Number} A 32bit integer (can be negative)
+ * @see http://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/
+ */
+function hashCode(str) {
+    let hash = 0;
+    for (let i = 0, len = str.length; i < len; i++) {
+        const chr = str.charCodeAt(i);
+        hash = ((hash << 5) - hash) + chr;
+        hash |= 0; // Convert to 32bit integer
```

Comment:
hash is already integer no?

Reply:
It is already an integer, but it could be using more than 32 bits. Bitwise operations in javascript seem to just return numbers of maximum 32 bits. The `| 0` in this case is dropping all bits passed the 32nd bit. It may be here because the original java code this was derived from expected to return an 32 bit integer hash.

I didn't change it because I don't have a strong reason to do that. This is a very used hash function, and I'm not sure about the implications of removing that piece of code.